homeassistant/components/modem_callerid/config_flow.py

```diff
@@ -0,0 +1,71 @@
+"""Config flow for Modem Caller ID integration."""
+from phone_modem import PhoneModem, exceptions
+import serial
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_DEVICE, CONF_NAME
+
+from . import _LOGGER
+from .const import DEFAULT_DEVICE, DEFAULT_NAME
+from .const import DOMAIN  # pylint:disable=unused-import
+
+DATA_SCHEMA = vol.Schema({"name": str, "device": str})
+
+
+class PhoneModemFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Phone Modem."""
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is not None:
+            name = user_input[CONF_NAME]
```

Comment:
Name isn't needed to setup this device, we can remove it.

They can edit the title of the config entry later if they want to name is something custom.